src/mlpack/tests/main_tests/approx_kfn_test.cpp

```diff
@@ -0,0 +1,396 @@
+{
+  arma::mat referenceData;
+  referenceData.randu(2, 80); // 80 points in 2 dimensions.
+
+  // First setting.
+  SetInputParam("reference", std::move(referenceData));
+  // Random input, k <= reference points.
+  SetInputParam("k", (int) 5);
+  // Random input, num_tables > 0.
+  SetInputParam("num_projections", (int) 4);
+  SetInputParam("num_tables", (int) 3);
+  // First solution.
+  mlpackMain();
+
+  // Get the distances matrix after first training.
+  arma::mat FirstOutputDistances =
+      std::move(CLI::GetParam<arma::mat>("distances"));
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+  // Second setting.
+  referenceData.randu(2, 80); // 80 points in 2 dimensions.
+  SetInputParam("reference", std::move(referenceData));
+  // Same input as first setting, k <= reference points.
+  SetInputParam("k", (int) 5);
+  // Random input, num_tables > 0.
+  SetInputParam("num_projections", (int) 6);
+  SetInputParam("num_tables", (int) 3);
+
+  // Second solution.
+  mlpackMain();
+
+  // Get the distances matrix after second training.
+  arma::mat SecondOutputDistances =
+      std::move(CLI::GetParam<arma::mat>("distances"));
+
+  // Check that the size of distance matrices (FirstOutputDistances and
+  // SecondOutputDistances) are not equal which ensures num_tables changes
+  // the output model.
+  CheckMatricesNotEqual(FirstOutputDistances, SecondOutputDistances);
+}
+
+/**
+* Make sure that the dimensions of the exact distances matrix are correct.
+ */
+BOOST_AUTO_TEST_CASE(ApproxKFNExactDistDimensionTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(2, 80); // 80 points in 2 dimensions.
+
+  SetInputParam("reference", std::move(referenceData));
+  // Random input, any k <= reference points.
+  SetInputParam("k", (int) 10);
+  SetInputParam("calculate_error", (bool) true);
+
+  // Random matrix specifying exact distances of each point to its k neighbors.
+  // Note that the values in the matrix do not matter as we are only concernec
+  // with the dimensions of the matrix passed.
+  arma::mat exactDistances;
+  exactDistances.randu(9, 90); // Wrong size (should be (1, 80)).
+  SetInputParam("exact_distances", std::move(exactDistances));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Make sure that the two strategie (Drusilla Select and QDAFN) output different
+   results.
+ */
+BOOST_AUTO_TEST_CASE(ApproxKFNDifferentAlgoTest)
+{
+  arma::mat referenceData;
+  referenceData.randu(6, 100); // 100 points in 6 dimensions.
+
+  SetInputParam("reference", std::move(referenceData));
+  // Random input, any k <= reference points.
+  SetInputParam("k", (int) 10);
+  SetInputParam("algorithm", (string) "ds");
+
+  // First solution.
+  mlpackMain();
+
+  // Get the distances and neighbors matrix after first training.
+  arma::mat FirstOutputDistances =
```

Comment:
As per the style guide, variables shouldn't have capitalized names, so could you please change this to `firstOutputDistances`?  (This applies a few other places.)